jobs: flake in TestRegistryLifecycle/rollback

Found in https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests/2178285?

```
------- Stdout: -------
=== RUN   TestRegistryLifecycle/rollback
--- FAIL: TestRegistryLifecycle/rollback (2.75s)
jobs_test.go:200: Starting resume
jobs_test.go:208: Exiting resume
jobs_test.go:200: Starting resume
jobs_test.go:208: Exiting resume
jobs_test.go:250: Starting success
jobs_test.go:254: Exiting success
jobs_test.go:200: Starting resume
jobs_test.go:648: expected job status: succeeded but got: running
jobs_test.go:208: Exiting resume
```

--- a/pkg/jobs/jobs_test.go
+++ b/pkg/jobs/jobs_test.go
@@ -34,7 +34,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/tests"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
@@ -545,8 +544,6 @@ func TestRegistryLifecycle(t *testing.T) {
 
 	// Verify that pause and cancel in a rollback do nothing.
 	t.Run("rollback", func(t *testing.T) {
-		skip.WithIssue(t, 52767, "flaky")
-
 		rts := registryTestSuite{}
 		rts.setUp(t)
 		defer rts.tearDown()

